src/data/pi/lex.lean

```diff
@@ -86,29 +95,31 @@ noncomputable instance [linear_order ι] [is_well_order ι (<)] [∀ a, linear_o
 @linear_order_of_STO (Πₗ i, β i) (<)
   { to_is_trichotomous := is_trichotomous_lex _ _ is_well_founded.wf } (classical.dec_rel _)
 
-lemma lex.le_of_forall_le [linear_order ι] [is_well_order ι (<)] [Π a, linear_order (β a)]
+lemma lex.le_of_forall_le [linear_order ι] [hwf : is_well_order ι (<)] [Π a, partial_order (β a)]
   {a b : lex (Π i, β i)} (h : ∀ i, a i ≤ b i) : a ≤ b :=
-le_of_not_lt (λ ⟨i, hi⟩, (h i).not_lt hi.2)
+or_iff_not_imp_left.2 $ λ hne,
+  let ⟨i, hi, hl⟩ := hwf.to_is_well_founded.wf.has_min {i | a i ≠ b i} (function.ne_iff.1 hne) in
```

Comment:
```suggestion
  let ⟨i, hi, hl⟩ := ‹is_well_order _ _›.to_is_well_founded.wf.has_min {i | a i ≠ b i} (function.ne_iff.1 hne) in
```

Reply:
[This was actually merged](https://github.com/leanprover-community/mathlib/commit/5c88020dbba7f4ccc22562b7bd4a2b84fd2e5057#diff-7afc24d61cb458dafb7dc06846ecaa2dbbc235bceb031d3b2fee998e806f21abR89-R93) in #16740. However `is_well_founded.wf` simply works without the need to name `hwf`, so I'll change it in #16777.